File: src/main/java/io/github/huacnlee/autocorrectIdeaPlugin/FormatAction.java
package io.github.huacnlee.autocorrectIdeaPlugin;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.CommonDataKeys;
import com.intellij.openapi.editor.Editor;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NotNull;

public class FormatAction extends AnAction {
    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        final Project project = e.getProject();
        final Editor editor = e.getData(CommonDataKeys.EDITOR);
        if (editor == null) {
            return;
        }
        var doc = editor.getDocument();
        
        AutoCorrectExecutor.format(project, doc);
    }
}

File: src/main/java/io/github/huacnlee/autocorrectIdeaPlugin/FormatAction.kt
package io.github.huacnlee.autocorrectIdeaPlugin

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.CommonDataKeys

class FormatAction : AnAction() {
    override fun actionPerformed(e: AnActionEvent) {
        val project = e.project
        val editor = e.getData(CommonDataKeys.EDITOR)
        Formatter.format(project, editor!!.document)
    }
}